### PR TITLE
Set SurfaceProvider directly on PreviewViewModel

### PR DESCRIPTION
**Pull request by temcguir**
Rather than wait for a SurfaceProvider to start the camera, it
 is set eagerly on the ViewModel/CameraUseCase

 The old behavior of waiting for the SurfaceProvider, this was 
 masking a bug where the camera may not have finished
 initialization before starting, so starting the camera now 
 explicitly waits for initialization to be finished

### CODE REVIEW

**Comment by temcguir** [comment]
@JolandaVerhoef this PR accomplishes something similar to #110, but without the addition of the `LifecycleOwner` to `CameraXViewfinder`. It also now explicitly calls `startCamera`/`stopCamera` using the `LifecycleStartEffect` you introduced in #108. PTAL.

File: domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/test/FakeCameraUseCase.kt [hunk 1]
```diff
@@ -49,6 +50,9 @@ class FakeCameraUseCase(
     var recordingInProgress = false
 
     var isLensFacingFront = false
+
+    private var surfaceProvider: Preview.SurfaceProvider? = null
```
Comment (JolandaVerhoef) [inline domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/test/FakeCameraUseCase.kt:54]: Not used

File: domain/camera/src/test/java/com/google/jetpackcamera/domain/camera/test/FakeCameraUseCaseTest.kt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2023 The Android Open Source Project
+ * Copyright (C) 2023-2024 The Android Open Source Project
```
Comment (JolandaVerhoef) [inline domain/camera/src/test/java/com/google/jetpackcamera/domain/camera/test/FakeCameraUseCaseTest.kt:2]: Replace with just 2024

**Review by JolandaVerhoef — APPROVED** [review]

File: domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/test/FakeCameraUseCase.kt (hunk 1 above)
Comment (temcguir) [reply domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/test/FakeCameraUseCase.kt]: Good catch. Removed.

File: domain/camera/src/test/java/com/google/jetpackcamera/domain/camera/test/FakeCameraUseCaseTest.kt (hunk 2 above)
Comment (temcguir) [reply domain/camera/src/test/java/com/google/jetpackcamera/domain/camera/test/FakeCameraUseCaseTest.kt:2]: This is spotless's doing. I'm going to turn off this behavior in a separate PR and we'll stick with the inception year in each file (as is normally done in Android).

**Pull request merged**